x-pack/plugins/entity_manager/server/lib/entity_client.ts

```diff
@@ -221,21 +225,30 @@ export class EntityClient {
     sources,
     start,
     end,
+    sortBy,
     metadataFields = [],
     filters = [],
     limit = 10,
   }: {
     sources: EntitySource[];
     start: string;
     end: string;
+    sortBy?: SortBy;
     metadataFields?: string[];
     filters?: string[];
     limit?: number;
   }) {
     const entities = await Promise.all(
       sources.map(async (source) => {
-        const mandatoryFields = [source.timestamp_field, ...source.identity_fields];
+        const mandatoryFields = [
+          ...source.identity_fields,
+          ...(source.timestamp_field ? [source.timestamp_field] : []),
+          ...(source.display_name ? [source.display_name] : []),
```

Comment:
I wonder if we can/should fill in the display_name with id if the display_name field or value is missing, WDYT? Else we leave that to the UI to manage which is also fine

UPDATE: I think you're already doing this with the COALESCE part?

Reply:
yes coalesce takes care of the fallback